### PR TITLE
PAL-999AMS, removed common import and replaced it with common code, u…

### PR DESCRIPTION
**Pull request by developer01189998819991197253**
…pdated imports and tests

Moved token to common
renamed akkatest and dbtest yamls to akka-test and db-test yaml
Updated some broken, or outdated tests

### CODE REVIEW

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java [hunk 1]
```diff
@@ -15,90 +15,58 @@
...
-    }
-
     /**
-     * Grouped assertion test
      * Create the object with the builder and then convert to the Json equivalent.
      * Takes the JSON Object, deserialises and tests against the original Object
      *
-     * @throws IOException throws if the {@link AttributeMaskingRequest} object cannot be converted to a JsonContent.
-     *                     This equates to a failure to serialise or deserialise the string.
+     * @throws JsonProcessingException throws if the {@link AttributeMaskingRequest} object cannot be converted to a JsonContent.
```
Comment (m78233) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:40]: Statement needs rewording.  JsonProcessingException can be thrown for either the writeValueAsString or the readValue, i.e. of either an issue with serialising or deserialising.   Also, the reference to a JsonContent object is no longer valid.  This test no longer uses this Spring object.

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingResponseTest.java [hunk 2]
```diff
@@ -15,82 +15,53 @@
...
-        assertThat(jsonTester).isNotNull();
-    }
-
     /**
-     * Grouped assertion test
      * Create the object with the builder and then convert to the Json equivalent.
      * Takes the JSON Object, deserialises and tests against the original Object
      *
-     * @throws IOException throws if the {@link AttributeMaskingResponse} object cannot be converted to a JsonContent.
+     * @throws JsonProcessingException throws if the {@link AttributeMaskingResponse} object cannot be converted to a JsonContent.
```
Comment (m78233) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingResponseTest.java:38]: See previous comment about JsonContent and about the name of the test.

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AuditErrorMessageTest.java [hunk 3]
```diff
@@ -15,86 +15,56 @@
...
-    }
 
     /**
-     * Grouped assertion test
      * Create the object with the builder and then convert to the Json equivalent.
      * Takes the JSON Object, deserialises and tests against the original Object
      *
-     * @throws IOException throws if the {@link AuditErrorMessage} object cannot be converted to a JsonContent.
-     *                     This equates to a failure to serialise or deserialise the string.
+     * @throws JsonProcessingException throws if the {@link AuditErrorMessage} object cannot be converted to a JsonContent.
```
Comment (m78233) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AuditErrorMessageTest.java:37]: See previous comment about JsonContent and the name of the test

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java (hunk 1 above)
Comment (m78233) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:40]: Minor point, the tests have been simplified down to the level that they are not grouped tests anymore and so the name of the test does not match with that it does.  Could refactor the name of tests, for example this one could be changed to testAttributeMaskingRequestSerialisingAndDeserialising.  Generalisation that could be applied to each of the test that have been refactored.

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java [hunk 4]
```diff
@@ -15,90 +15,58 @@
...
-import org.springframework.test.context.ContextConfiguration;
 
-import uk.gov.gchq.palisade.Context;
-import uk.gov.gchq.palisade.User;
-import uk.gov.gchq.palisade.resource.LeafResource;
-import uk.gov.gchq.palisade.resource.impl.FileResource;
-import uk.gov.gchq.palisade.resource.impl.SystemResource;
-import uk.gov.gchq.palisade.rule.Rules;
-import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
+import uk.gov.gchq.palisade.service.attributemask.common.Context;
```
Comment (m78233) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:22]: The package name common, is this an intermediate step in the refactoring or the final destination?  Some of these objects could be better suited for existing packages such as User could be moved to the model package.

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/web/RestControllerWebMvcTest.java [hunk 5]
```diff
@@ -30,7 +30,7 @@
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
 
 import uk.gov.gchq.palisade.service.attributemask.ApplicationTestData;
-import uk.gov.gchq.palisade.service.attributemask.model.Token;
+import uk.gov.gchq.palisade.service.attributemask.common.Token;
```
Comment (m78233) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/web/RestControllerWebMvcTest.java:37]: There is a Token class in the package model as well as the package common.   One needs to be removed, my vote would be the one in common.

File: attribute-masking-service/src/contract-tests/resources/application-akka-test.yaml [hunk 6]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2018-2021 Crown Copyright
```
Comment (m78233) [inline attribute-masking-service/src/contract-tests/resources/application-akka-test.yaml]: Looks like application-akklatest.yaml and application-dbtest.yaml are still in the repo and should be removed.

File: attribute-masking-service/src/unit-tests/java/uk/gov/gchq/palisade/service/attributemask/common/ContextTest.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        var actualJson = mapper.writeValueAsString(context);
+        var actualInstance = mapper.readValue(actualJson, Context.class);
+
+        // Then
+        assertThat(actualInstance)
+                .as("Check using recursion, that the %s has been deserialized successfully", context.getClass().getSimpleName())
+                .usingRecursiveComparison()
+                .isEqualTo(context);
+    }
+}
```
Comment (m78233) [inline attribute-masking-service/src/unit-tests/java/uk/gov/gchq/palisade/service/attributemask/common/ContextTest.java:44]:
```suggestion
}

```

**Review by m78233 — CHANGES_REQUESTED** [review]
Minor suggestions and a question.

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/web/RestControllerWebMvcTest.java (hunk 5 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/web/RestControllerWebMvcTest.java:37]: updated!

File: attribute-masking-service/src/contract-tests/resources/application-akka-test.yaml (hunk 6 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/contract-tests/resources/application-akka-test.yaml]: removed, good spot

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java (hunk 4 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:22]: I think if they are external to the module, and were previously dependencies, they should remain in common.
If they change and become module specific then they can be pulled into each modules model package

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AuditErrorMessageTest.java (hunk 3 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AuditErrorMessageTest.java:37]: updated

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingResponseTest.java (hunk 2 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingResponseTest.java:38]: updated

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java (hunk 1 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:40]: updated

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java (hunk 4 above)
Comment (m78233) [reply attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/model/AttributeMaskingRequestTest.java:22]: Sounds good

**Review by m78233 — DISMISSED** [review]
Looks good to me

**Pull request closed without merging**